### PR TITLE
Use ctrl syntax and explicit filtering of empty labels.

### PR DESCRIPTION
**Pull request by akash1810**
Helps address #902 from the client-side.

### CODE REVIEW

File: kahuna/public/js/services/label.js [hunk 1]
```diff
@@ -13,16 +13,20 @@ labelService.factory('labelService', ['$rootScope', '$q', function ($rootScope,
                 .then(newLabels => {
                     image.data.userMetadata.data.labels = newLabels;
                     $rootScope.$emit('image-updated', image, image);
+                    return image;
                 });
         }
     }
 
     function add (image, labels) {
+        labels = labels.filter(label => label.trim().length > 0);
```
Comment (akash1810) [inline kahuna/public/js/services/label.js]: Filter out all empty strings (don't trust anything...)

**Comment by jamesgorrie** [comment]
Does this make sense - what if I want to apply none to all?

File: kahuna/public/js/services/label.js (hunk 1 above)
Comment (jamesgorrie) [inline kahuna/public/js/services/label.js]: You could filter on `!` which would remove `undefined`s etc

**Comment by akash1810** [comment]
You mean remove all labels? Why would you want to here? When you upload, there are no labels. So presumably, if you've applied a label(s) you did so on purpose?

**Comment by jamesgorrie** [comment]
> did so on purpose

Or hit apply to all by accident. Every action should have some form of redo - apply none to all would be the antithesis of apply to all.

**Comment by akash1810** [comment]
> apply none to all would be the antithesis of apply to all.

I was planning on replicating the behaviour of the batch selection labeller:

screen shot 2015-07-10 at 10 47 53

So you could apply/remove a label on a per label basis. Thoughts?

**Comment by theefer** [comment]
Trying to unify the batch systems in the future makes sense.

**Comment by jamesgorrie** [comment]
Could we move your label stuff over instead of this?

**Comment by akash1810** [comment]
Just checked - apply none to all doesn't work atm...

> Could we move your label stuff over instead of this?

Yep - will confirm the UX with @alastairjardine beforehand though.

Will change this PR to be just about the filter.

**Comment by theefer** [comment]
:+1:

**Pull request merged**